### DISCUSSION TITLE
Multiple timeframes

### DISCUSSION THREAD

**Discussion by tonyugabriel** [edtechre/pybroker#164] [General]
Wondering if there is a way to use multiple timeframes at once.  For example providing 5 minute resolution bars but be able to write logic on 1hour bars?  I can code around this but wondering if there is already such a feature or plans for one?

**Comment by edtechre** [comment]
Hi @tonyugabriel,

Do you mean running an execution for every 1 hour on 5 minute bars, or do you mean somehow aggregating the 5 min bars into 1 hour bars before running an execution?

**Reply by tonyugabriel** [reply]
Hi, 

My thinking is I would like the trade executions to run on 5minute data so the stop loss / take profit would be more acuurate as to exactly when you got stopped out, but the logic i use to decide on taking a position would be on 1 hour bars.  So for example lets take a simple moving cross over stratey  with a sl/tp, that would be calculated off 1 hour bars but when I then take a position the engine is using 5 minute bars to decide when the sl/tp is hit.  I could make a custom data source that aggregates the 5min bars into hourly as extra columns but wondering if there is a way to provid both sets of bars to the engine itself.

**Reply by edtechre** [reply]
Hi @tonyugabriel,

Thank you for the explanation. I will look into adding this feature.

> I could make a custom data source that aggregates the 5min bars into hourly as extra columns but wondering if there is a way to provid both sets of bars to the engine itself.

Yes, in the meantime this is the same solution I would suggest.

**Reply by ghost** [reply]
Would this also work for Higher timeframe indicators would be a most welcome feature!
